AuthenticatorShared/UI/Platform/Application/Extensions/StringTests.swift

```diff
@@ -14,6 +15,15 @@ class StringTests: AuthenticatorTestCase {
         XCTAssertEqual("9c303aee-e636-4760-94b6-e4951d7b0abb".hashColor.description, "#C96CD2FF")
     }
 
+    /// `hexSHA256Hash` returns a hexadecimal string with a SHA-256 hash of the string.
+    func test_hexSHA256Hash() {
+        let subject = "String to be hashed"
+        let hashedData = SHA256.hash(data: Data(subject.utf8))
+        let expected = hashedData.map { String(format: "%02hhx", $0) }.joined()
+
+        XCTAssertEqual(subject.hexSHA256Hash, expected)
```

Comment:
⛏️  IMO when testing hashes or encrypted stuff is better to just put the hardcoded expected string as to do the procedure; something like a black box flow, you enter the string and you just compare the output with the expected one. By doing the comparison against the hardcoded expected string you could additionally detect internal library changes if let's say `SHA256.hash` has a bug in the future or things like that.

Reply:
I like that approach 👍 Update in latest to use a hardcoded string.